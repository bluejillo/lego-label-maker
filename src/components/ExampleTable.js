import styled from 'styled-components';

const Table = styled.table`
    border-collapse: collapse;
    margin-top: 0.5rem;
`;
const Thead = styled.thead`
    background: darkgrey;
    font-weight: 500;
`;
const Th = styled.th`
    border: solid 1px black;
    padding: 0.5rem;
`;
const Td = styled.td`
    border: solid 1px black;
    padding: 0.5rem;
`;
export const ExampleTable = () => {
    return (
        <Table>
            <Thead>
                <tr>
                  <Th>PartID</Th>
                  <Th>PartName</Th>
                  <Th>GroupID</Th>
                </tr>
            </Thead>
            <tbody>
                <tr>
                  <Td>123</Td>
                  <Td>4 stud brick</Td>
                  <Td></Td>
                </tr>
                <tr>
                  <Td>222</Td>
                  <Td>1 stud brick</Td>
                  <Td>1</Td>
                </tr>
                <tr>
                  <Td>333</Td>
                  <Td>1 stud flat brick</Td>
                  <Td>1</Td>
                </tr>
                <tr>
                    <Td>555</Td>
                    <Td>transparent stud</Td>
                    <Td></Td>
                </tr>
                <tr>
                  <Td>456</Td>
                  <Td>6 stud brick</Td>
                  <Td>2</Td>
                </tr>
                <tr>
                  <Td>775</Td>
                  <Td>2x6 stud brick</Td>
                  <Td>2</Td>
                </tr>
            </tbody>
        </Table>
    );
};