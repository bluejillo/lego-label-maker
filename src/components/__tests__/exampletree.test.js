import renderer from 'react-test-renderer';
import { ExampleTable } from '../ExampleTable';
import { cleanup } from '@testing-library/react';

afterEach(() => {
    cleanup();
});

test('Example table matches snapshot', () => {
    const table = renderer.create(<ExampleTable/>).toJSON();
    expect(table).toMatchSnapshot();
});