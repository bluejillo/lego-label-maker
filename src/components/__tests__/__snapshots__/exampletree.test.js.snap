// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example table matches snapshot 1`] = `
<table
  className="sc-bcXHqe eHJstJ"
>
  <thead
    className="sc-gswNZR bjjIGh"
  >
    <tr>
      <th
        className="sc-dkrFOg llaFUg"
      >
        PartID
      </th>
      <th
        className="sc-dkrFOg llaFUg"
      >
        PartName
      </th>
      <th
        className="sc-dkrFOg llaFUg"
      >
        GroupID
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        123
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        4 stud brick
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      />
    </tr>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        222
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        1 stud brick
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        1
      </td>
    </tr>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        333
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        1 stud flat brick
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        1
      </td>
    </tr>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        555
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        transparent stud
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      />
    </tr>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        456
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        6 stud brick
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        2
      </td>
    </tr>
    <tr>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        775
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        2x6 stud brick
      </td>
      <td
        className="sc-hLBbgP eGqmFI"
      >
        2
      </td>
    </tr>
  </tbody>
</table>
`;
